### PR TITLE
JSONB (and more) documentation

### PR DESCRIPTION
**Pull request by mickhansen**
WIP

### CODE REVIEW

File: docs/docs/models.md [hunk 1]
```diff
@@ -841,6 +844,38 @@ To recap&comma; the elements of the order &sol; group array can be the following
...
+
+```js
+sequelize.define('User', {}, {
+  indexes: [
+    // Create a unique index on email
+    {
+      unique: true,
+      fields: ['email']
+    },
+
```
Comment (janmeier) [inline docs/docs/models.md]: You are missing `method`, e.g. 
```js
{
                name: 'title_index',
                method: 'BTREE',
                fields: ['title']
              }
```

Comment (janmeier) [inline docs/docs/models.md]: And advanced fields
```js
fields: ['fieldB', {attribute: 'fieldA', collate: 'en_US', order: 'DESC', length: 5}]
```

Comment (mickhansen) [inline docs/docs/models.md]: Yeah best cover that too, i was a bit too lazy coming up with every single case and figured i'd write some cross cases for demonstration :D

File: docs/docs/querying.md [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+      }
+    }
+  ]
+}
+// title LIKE 'Boat%' OR description LIKE '%boat%'
+```
+
+### JSONB
+
+JSONB can be queried in three different ways.
```
Comment (janmeier) [inline docs/docs/querying.md]: Maybe show the resulting SQL here

**Pull request merged**